### ISSUE TITLE
Nuget publish will mistakenly publish if the build fails

### ISSUE THREAD

**Issue by wintondeshong** [AndcultureCode/AndcultureCode.Cli#62]
*Problem*
When running the `nuget --publish <version>` command I experienced an I/O error in creating the assets. Even though this occurred, the process kept going. It created the nupkg just fine so in my case, it ended up sending up the previously built assembly version with the new nupkg file. 

*Solution*
While the nuget pack commands does attempt to do a build, perhaps we consider doing an explicit dotnet build ourselves and error check that.

**Comment by brandongregoryscott** [comment]
@wintondeshong What's an easy way to reproduce this? Intentionally breaking the build thru erroneous syntax, or opening one of the files in an editor to lock the file?

Looks like the [`dotnet pack`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-pack?tabs=netcore2x) command provides `--no-build` and `--no-restore` flags which would allow us to have control over when the build occurs.

**Comment by wintondeshong** [comment]
@brandongregoryscott yeah, without having a chance to confirm, I'd probably try in that order (1) break build (2) lock file by opening in editor (3) change permissions of 'bin' directory to not have write access so it would throw the I/O error

When/ if I have a chance before this is worked on, I'll try to 100% verify the above.
